src/openssl.c

```diff
@@ -3427,10 +3435,25 @@ static BIO *getbio(lua_State *L) {
 } /* getbio() */
 
 
+static int pem_pw_cb(char *buf, int size, int rwflag, void *u) {
+	lua_State *L = (lua_State *) u;
+
+	if (lua_isnil(L, -1) || (lua_isfunction(L, -1) && lua_pcall(L, 0, 1, 0)))
+		return 0;
```

Comment:
This doesn't keep the stack balanced: if the field is nil then nothing is pushed. If the `pcall` results in an error then the error will be left on the stack.

Reply:
I suppose this should work?
```diff
diff --git a/src/openssl.c b/src/openssl.c
index 35183ab..868a6d3 100644
--- a/src/openssl.c
+++ b/src/openssl.c
@@ -4073,9 +4073,14 @@ static BIO *getbio(lua_State *L) {
 static int pem_pw_cb(char *buf, int size, int rwflag, void *u) {
        lua_State *L = (lua_State *) u;
 
-       if (lua_isnil(L, -1) || (lua_isfunction(L, -1) && lua_pcall(L, 0, 1, 0)))
+       if (lua_isnil(L, -1))
                return 0;
 
+       if (lua_isfunction(L, -1) && lua_pcall(L, 0, 1, 0)) {
+               lua_pop(L, 1);
+               return 0;
+       }
+
        const char *pass = lua_tostring(L, -1);
        if (!pass)
                return 0;
```